internal/service/service_test.go

```diff
@@ -0,0 +1,170 @@
+}
+
+// Unary calls are easy to test!
+// Just call them directly
+func TestUnaryCalls(t *testing.T) {
+	ctx := context.Background()
+	mockUserGetter := &mockUserGetter{user: "someuser"}
+	jobService := service.NewJobService(1, mockUserGetter, "/tmp")
+
+	t.Run("start-stop-status", func(tt *testing.T) {
+		resp, err := jobService.StartJob(ctx, &jobmanagerpb.StartJobRequest{
+			Command: echoPathRelative,
+			Args:    []string{"echo", "5"},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.NotNil(t, resp.JobId)
+
+		stopResp, err := jobService.StopJob(ctx, &jobmanagerpb.StopJobRequest{
+			JobId: resp.JobId,
+		})
+		require.NoError(t, err)
+		require.NotNil(t, stopResp)
+
+		time.Sleep(1 * time.Second)
+
+		statusResp, err := jobService.GetStatus(ctx, &jobmanagerpb.GetStatusRequest{
+			JobId: resp.JobId,
+		})
+		require.NoError(t, err)
+		require.NotNil(t, statusResp)
+		require.Equal(t, jobmanagerpb.Status_STATUS_STOPPED, statusResp.CurrentStatus)
+	})
+
+	t.Run("invalid-user", func(tt *testing.T) {
+		// Create a job
+		resp, err := jobService.StartJob(ctx, &jobmanagerpb.StartJobRequest{
+			Command: echoPathRelative,
+			Args:    []string{"echo", "5"},
+		})
+		require.NoError(tt, err)
+		require.NotNil(tt, resp)
+		require.NotNil(tt, resp.JobId)
+
+		// Change the user
+		mockUserGetter.user = "anotheruser"
+
+		// Stopping the job should fail with "not found" since we are no longer
+		// the right user
+		stopResp, err := jobService.StopJob(ctx, &jobmanagerpb.StopJobRequest{
+			JobId: resp.JobId,
+		})
+		require.Error(tt, err)
+		stat, ok := status.FromError(err)
+		require.True(tt, ok)
+		require.Equal(tt, codes.NotFound, stat.Code())
+		require.Nil(t, stopResp)
+	})
+
+}
+
+// Streaming is a little more challenging
+// We could generate some mocks (I like github.com/maxbrunsfeld/counterfeiter)
+// But for basic black box tests, a local server is easy enough to spin up
+func TestService(t *testing.T) {
+	srv := testutils.GrpcLocalServer{}
+	jobService := service.NewJobService(1, &mockUserGetter{user: "someuser"}, "/tmp")
+	server := grpc.NewServer()
+
+	jobService.Register(server)
+	require.NoError(t, srv.ListenAndServe(server))
+	t.Cleanup(func() {
+		server.Stop()
+		// Not really interested in the error
+		// returned by the server. We just want to make sure
+		// it has cleaned up
+		_ = srv.Done()
+	})
+
+	ctx := context.Background()
+	jobClient := jobmanagerpb.NewJobManagerClient(srv.Conn())
+
+	t.Run("stream-stdout", func(tt *testing.T) {
+		resp, err := jobClient.StartJob(ctx, &jobmanagerpb.StartJobRequest{
+			Command: echoPathRelative,
+			Args:    []string{"echo", "5"},
+		})
+		require.NoError(tt, err)
+		require.NotNil(tt, resp)
+		require.NotNil(tt, resp.JobId)
+
+		outputclient, err := jobClient.GetJobOutput(ctx, &jobmanagerpb.GetJobOutputRequest{
+			JobId: resp.JobId,
+			Type:  jobmanagerpb.OutputType_OUTPUT_TYPE_STDOUT,
+		})
+		require.NoError(tt, err)
+
+		fullOutput := bytes.NewBuffer(nil)
```

Comment:
The zero-value works too, and then there's less questions about what the magical `nil` argument is for.

```suggestion
		var fullOutput bytes.Buffer
```